### DISCUSSION TITLE
Any examples of parser plugins?

### DISCUSSION THREAD

**Discussion by tmcw** [babel/babel#15364] [Q&A] [answered]
I'm interested in writing a small new plugin to support a particular, proprietary syntax - for example, variables that start with a new character, like `@foo`. It's very hard to find any examples of plugins that do this, or documentation of how it's done. As far as I can tell:

- All of the parser plugins listed on the site have changed so that they now enable options in the core parser. They don't contain any parsing code.
- I can't find any documentation for implementing parsers on the site. It looks like some of the internal parsers are implemented as mixins on the `Parser` class, like you would do with `acorn`, but it's not clear how this is accessible from an external parser.

Thanks for any help!

**Comment by liuxingbaoyu** [comment, marked as answer]
This is a bit difficult.
Babel parser currently does not provide compatibility guarantees internally, and with the support of new syntax, external plugins may be broken at any time.
And currently I don't know of any parser plugins other than forked repositories.

Maybe you can take a look at this.
https://github.com/acornjs/acorn#plugin-developments
